lib/Sema/TypeCheckDecl.cpp

```diff
@@ -2653,6 +2611,287 @@ findNonImplicitRequiredInit(const ConstructorDecl *CD) {
+          // initStorage(Value) -> Storage.
+          if (!foundFunc->isStatic())
+            continue;
+          auto methodTy = foundFunc->getInterfaceType()
+            ->castTo<AnyFunctionType>()
+            ->getResult();
+          if (!methodTy->isEqual(expectedInitStorageTy))
+            continue;
+          
+          if (initStorageDecl) {
+            TC.diagnose(behavior->getLoc(),
+                        diag::property_behavior_protocol_reqt_ambiguous,
+                        TC.Context.Id_initStorage);
+            TC.diagnose(initStorageDecl->getLoc(),
+                        diag::property_behavior_protocol_reqt_here,
+                        TC.Context.Id_initStorage);
+            TC.diagnose(foundFunc->getLoc(),
+                        diag::property_behavior_protocol_reqt_here,
+                        TC.Context.Id_initStorage);
+            break;
+          }
+          
+          initStorageDecl = foundFunc;
+        }
+      }
+      
+      if (!initStorageDecl) {
+        TC.diagnose(behavior->getLoc(),
+                    diag::property_behavior_protocol_no_initStorage,
+                    expectedInitStorageTy);
+        for (auto found : lookup)
+          TC.diagnose(found.Decl->getLoc(),
+                      diag::found_candidate);
+      }
+      
+      // TODO: Support storage-backed behaviors in non-type contexts.
+      if (!dc->isTypeContext()) {
+        TC.diagnose(behavior->getLoc(),
+                    diag::property_behavior_with_storage_not_supported,
+                    behaviorProto->getName());
+        conformance->setInvalid();
+        continue;
+      }
+      
+      // Instantiate the storage next to us in the enclosing scope.
+      TC.completePropertyBehaviorStorage(decl, storageReqt,
+                                         initStorageDecl,
+                                         behaviorSelf,
+                                         storageTy,
+                                         conformance,
+                                         interfaceSubs,
+                                         contextSubs);
+      continue;
+    }
+    
+    if (auto func = dyn_cast<FuncDecl>(requirement)) {
+      // Handle accessors as part of their property.
+      if (func->isAccessor())
+        continue;
+    }
+
+    unknownRequirement(requirement);
+  }
+  
+  // Bail out if we didn't resolve method witnesses.
+  if (conformance->isInvalid()) {
+    decl->setInvalid();
+    decl->overwriteType(ErrorType::get(TC.Context));
+    return;
+  }
+
+  conformance->setState(ProtocolConformanceState::Complete);
+
+  // Check that the 'value' property from the protocol matches the
+  // declared property type in context.
+  auto sig = behaviorProto->getGenericSignatureOfContext();
+  auto map = sig->getSubstitutionMap(interfaceSubs);
+  auto substValueTy = behavior->ValueDecl->getInterfaceType()
+    .subst(decl->getModuleContext(), map, SubstOptions());
```

Comment:
Doesn't seem to be. I was just following Xcode code completion, which swallows defaulted args.